### PR TITLE
WebAssembly Support

### PR DESCRIPTION
**Pull request by kateinoigakukun**
- 2548c5e The main change of this PR adds WebAssembly/WASI variant as platform.
- 6d16075 This change is necessary to make swift-autolink-extract work.
For wasm target, swift-autolink-extract try to extract info from swiftmodule. But without target argument, compiler emits module as Mach-O format on macOS which is not supported by swift-autolink-extract. So SwiftPM need to pass target option when emitting swiftmodule.

This is a part of SR-9307 and [#24684](https://github.com/apple/swift/pull/24684).

### CODE REVIEW

File: Sources/PackageDescription/SupportedPlatforms.swift [hunk 1]
```diff
@@ -41,6 +41,10 @@ public struct Platform: Encodable {
     /// The Android platform
     @available(_PackageDescription, introduced: 5.2)
     public static let android: Platform = Platform(name: "android")
+
+    /// The WASI platform
+    @available(_PackageDescription, introduced: 5.2)
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription/SupportedPlatforms.swift]: Can you make this 999.0 as master is no longer tracking Swift 5.2. We'll replace all APIs with 999.0 when the version of Swift is announced.

**Comment by aciidb0mb3r** [comment]
LGTM, thank you for working on this! I just have one minor comment about the availability of the new API.

File: Sources/PackageDescription/SupportedPlatforms.swift (hunk 1 above)
Comment (jakepetroules) [reply Sources/PackageDescription/SupportedPlatforms.swift]: Should we call the property `webAssembly` to be more discoverable?

(The string bit needs to stay as "wasi", though.)

Comment (aciidb0mb3r) [reply Sources/PackageDescription/SupportedPlatforms.swift]: that would be fine with me but maybe we should look at some other build system/package manager that supports web assembly and see what they're using as the public facing name?

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by kateinoigakukun** [comment]
Sorry, I fixed test cases to fit this change 🙏

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by kateinoigakukun** [comment]
I'm not sure why the smoke test on Linux failed 🤔 
It seems to fail while building test suites. Is this related to cache?

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test linux

**Comment by kateinoigakukun** [comment]
Passed 🎉

**Pull request merged**